src/NServiceBus.Core/Pipeline/Outgoing/InspectOutgoingMessageBehavior.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    class InspectOutgoingMessageBehavior : Behavior<IOutgoingPhysicalMessageContext>
```

Comment:
would the conflict with the audit headers be solved if we correctly clone headers [here](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Audit/InvokeAuditPipelineBehavior.cs#L21) instead of passing the reference?